src/pages/ProductListing.js

```diff
@@ -0,0 +1,60 @@
+import { Navbar } from "../components/Navbar"
+import { Footer } from "../components/Footer"
+import "../styles/pages/productlist.css"
+import { FiltersBar } from "../components/FiltersBar"
+import { ProductCard } from "../components/ProductCard"
+import {useState,useEffect} from "react";
+import axios from "axios"
+import { useProduct } from "../context/product-context"
+import {filterProducts} from "../utils/filterProducts"
+
+
+
+export const ProductListing = ()=>{
+    const [loader, setLoader] = useState(true);
+    const [products, setProducts] = useState([{}]);
+    const {state} = useProduct();
+
+    useEffect(() => {
+      (async function prodLoader() {
+        try {
+          const response = await axios.get("/api/products");
+          const getProducts = response.data.products;
+          setLoader(false);
+          setProducts(getProducts);
+        } catch (error) {
+          console.error(error);
+        }
+      })(); //IIFE - Immediately Invoked Function
+    }, []);
+
+
+    const displayProducts = filterProducts(state,products);
+    console.log("dp",displayProducts);
+    console.log("Display State",state);
+    return(
+        <>
+        <Navbar/>
+      
+     <div className="product-container-main">
+
+        {/* Filters Side-Bar  */}
+        <FiltersBar/>
+        {/* Product Cards  */}
+        <div className="fs-md">{loader && "Loading Products..."}</div>
```

Comment:
I like how you are handling each state here

Reply:
Thank you.